### PR TITLE
execNewBlock: remove the parent header as parameter and sync from the db instead

### PR DESCRIPTION
**Pull request by ak3n**
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205566925523372

### CODE REVIEW

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs
Comment (edmundnoble) [inline test/Chainweb/Test/Pact/PactMultiChainTest.hs:1423]: No need for these parens

**Review by edmundnoble — COMMENTED** [review]
This can't be merged until the release. All of my comments are pedantic stuff which barely matters, I'm extremely impressed that we can get away with this, it looks exactly like I'd expect. We need to test this on a real mining node as well before mergew.

File: test/Chainweb/Test/Pact/PactMultiChainTest.hs
Comment (edmundnoble) [reply test/Chainweb/Test/Pact/PactMultiChainTest.hs:1425]: Hell maybe this should be a shared utility.

**Comment by edmundnoble** [comment]
We actually don't really need to test this on a real mining node, fake mining with a simulated time distribution also works, as long as it's in a network with multiple mining nodes.

**Comment by ak3n** [comment]
@edmundnoble will wait until the master is fixed so we can merge with green CI this PR? Or can merge already? it was green before

**Comment by edmundnoble** [comment]
@ak3n I want this PR benchmarked to see how many blocks we produce with a simulated miner first. I can help with that some time when you're free.

**Comment by edmundnoble** [comment]
Actually we should probably use a load test for this benchmark.

File: bench/Chainweb/Pact/Backend/ForkingBench.hs
Comment (0xd34df00d) [inline bench/Chainweb/Pact/Backend/ForkingBench.hs:135]: You might create a helper function (for the tests only!) like
```haskell
updateParentHeader join pwo queue = void $ validateBlock join (payloadWithOutputsToPayloadData pwo) queue
```
avoiding the comments in the tests and making the tests a bit more expressive.

File: src/Chainweb/Pact/PactService.hs
Comment (0xd34df00d) [inline src/Chainweb/Pact/PactService.hs:296]: Tiny nit: this `do` isn't necessary.

File: src/Chainweb/Pact/Service/Types.hs
Comment (0xd34df00d) [inline src/Chainweb/Pact/Service/Types.hs:298]: The parens are probably no longer needed.

**Review by 0xd34df00d — APPROVED** [review]
The logic looks good as per discussion, so I only have a few minor nits.

**Comment by ak3n** [comment]
Merged in #1803

**Pull request closed without merging**
